app/controllers/posts_controller.rb

```diff
@@ -7,4 +7,26 @@ def index
   def show
     @post = Post.find(params[:id])
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = current_user.posts.build(post_params)
+
+    if @post.save
+      flash[:success] = 'Post created successfully'
+      redirect_to user_posts_path(current_user)
+    else
+      flash.now[:error] = 'Post creation failed'
+      render :new
+    end
+  end
+
+  private
+
+  def post_params
+    params.require(:post).permit(:title, :text)
+  end
 end
```

Comment:
## Important as this will affect future milestones ⚠️ 
- [ ] This requirement was missed by the previous reviewer.
The error message you're encountering indicates that the new action in your PostsController is missing a template for the text/html format. This error occurs because Rails expects an action to render a template with the same name as the action
If you want the new action to render the HTML template for creating a post, rename `next.html.erb` to `new.html.erb` in the app/views/posts folder.
![image](https://github.com/zamcham/ThoughtTrail/assets/81932478/986ad88a-17d9-4dfe-89d7-46f36f745a3e)